doc/rst/mason-packages/guide/gitdependencies.rst

```diff
@@ -0,0 +1,29 @@
+:title: Mason Guide: *Git Dependencies*
+
+Specifying Dependencies from `git` Repositories
+===============================================
+To depend on a library located in a git repository, a git key pointing to the repository URL is all that is required:
+
+.. code-block:: text
+
+
+    [dependencies]
+    HelloWorld = { git = "https://github.com/bmcdonald3/HelloWorld" }
+
+
+The repository that is specified must be a mason package itself, but it does not need to be contained in a mason registry.
+
+Without specifying the branch to be used, a default branch of `master` will be assumed and if there is no master branch, the dependency will not work. To specify a different branch, the `branch` key can be used:
```

Comment:
We still check master, since that is still the default of repos created with git. This sprint we are trying to update this to use `HEAD` instead of `master`, but it is still work in progress.